src/screens/NavBarScreen/Profile/index.tsx

```diff
@@ -35,18 +43,27 @@ export const Profile = () => {
   const [currentUser, isCurrentUserLoading] = useAuthState(fireAuth);
   const [updateEmail] = useUpdateEmail(fireAuth);
   const [updatePassword] = useUpdatePassword(fireAuth);
+  const [refreshing, setRefreshing] = useState(false);
 
   const [userDoc, isUserDataLoading] = useDocument<User>(
-    doc(fireStore, 'users', currentUser?.uid ?? '') as DocumentReference<User>,
+    getUserRefByUid(currentUser?.uid ?? '') as DocumentReference<User>,
     {snapshotListenOptions: {includeMetadataChanges: true}},
   );
 
+  const onRefresh = React.useCallback(() => {
+    setRefreshing(true);
+    setTimeout(() => {
+      setRefreshing(false);
+    }, 2000);
+  }, []);
+
```

Comment:
I suppose you wanted to track the loading state of the update query. Then you need to do it a bit differently. 

Please, change the refresh control and add a custom state called `isUpdating`.
```
<RefreshControl refreshing={isCurrentUserLoading || isUpdating || isUserDataLoading} />

const updateUser = () => {
   setIsUpdating(true)
   
   ...
   
   setIsUpdating(false)
 }
```
You can even move it to the separate hook, but it's up to you.

Also, there is no need to call refetching methods for user documents, as they behave as observables. It means once the data has changed on firestore, you will immediately receive the changes in your component.